### DISCUSSION TITLE
How do I render and syntax-highlight code in a Python file?

### DISCUSSION THREAD

**Discussion by cpcloud** [quarto-dev/quarto-cli#6618] [Q&A] [answered]
### Description

I'd like to be able to do something analogous to `cat`ing a file and having the contents of it syntax highlighted.

In IPython or Jupyter, I can use the `%pycat` cell magic to do this.

How do I achieve the same thing in quarto?

I'm imagining something like:

````markdown
 ```{python source=filename.py}
 # highlighted contents of filename.py
 ```
````

**Comment by cpcloud** [comment, marked as answer]
Looks like I can do this with `Markdown`:

![image](https://github.com/quarto-dev/quarto-cli/assets/417981/c0b64fb7-5bef-4476-ba20-4e76fac5cfa4)

**Reply by mcanouil** [reply]
Yep, that's the way I would have suggested.

**Comment by cderv** [comment]
I believe the `include-code-files` extension can be useful here: https://github.com/quarto-ext/include-code-files

You could write something close to what you thought was possible 
````markdown
 ```{.python include="filename.py"}
 ```
````

Hope it helps

**Reply by cpcloud** [reply]
Thanks! I did see that too, but ultimately decided to avoid the extra dependency since `print` with `output: asis` works well!
